### ISSUE TITLE
Provide support for multiple servers instead of one

### ISSUE THREAD

**Issue by iamluca** [kyb3r/modmail#99]
self explanatory

**Comment by fourjr** [comment]
Can you elaborate on why this feature would be useful? Use cases

**Comment by sircam191** [comment]
Why not just make different bots for each server?

**Comment by kyb3r** [comment]
Modmail is meant to be used with only one/two servers since that way you know which server the member is messaging from. Currently, there are two server setup options:
- Using the same server as an inbox
- Using a separate server from the main guild as an inbox.

Both are supported. However, if you add the bot to more servers and the user that is messaging is not in the specified main guild, the thread created embed will not have member-related info such as roles etc. 
![image](https://user-images.githubusercontent.com/29310830/50819035-f93e6880-137c-11e9-9585-77e31eade54f.png)

So technically adding it to multiple servers is supported but is not exactly suitable for the uses of this bot.

**Comment by fourjr** [comment]
About the heroku issue:
1. Dyno Hours Limit - Modmail isn't affected by it
2. Apps Limit - Create a new account

**Comment by Taaku18** [comment]
I propose the enhancement to introduce segregation within the bot, not exactly a centralized server (although that can be an addiction). Where the bot wouldn’t be single server targeted, instead, all the configurations will be configured within the individual servers that the bot were deployed to. The backend database can also be modified accordingly, perhaps categorized by the server that the bot is serving. Different servers that the same bot is serving can have little or no connection with one another. If implemented correctly, there shouldn’t be any changes (other than you can use the same bot thread to manage multiple servers).

The bottom line I’m trying to make is that instead of making separate bots and hard coding config vars to setup the bots, the same task can be accomplished through dynamic addition of servers, server independent configurations, and segmentation of the database. All of that while consuming less resources.

**Comment by fourjr** [comment]
Config carries over if you use the same modmail api token

**Comment by kyb3r** [comment]
If a user shares multiple servers with the bot, there would be no way to know where the user is messaging from, and where to direct the message.

**Comment by Taaku18** [comment]
I guess that’s true, since the best one can do is get a list of all the servers the bot and the user has in common. You can, though, embed the information for all servers, and if you really need to know where they are from, you can always just ask. I see this might be considered going slightly different direction as you had planned with this bot. Still, I hope this gets considered in the future as the practicality of running separate apps for different servers is still questionable.

**Comment by fourjr** [comment]
How is it questionable? 🤔

Having different modmail bots for different servers is what you should do.

**Comment by kyb3r** [comment]
Having a dedicated bot for each server gives customisability that a lot of larger servers need. Also asking a user where they are from is not a streamlined user experience and isn't something I personally like.

**Comment by Taaku18** [comment]
Managing 50 servers isn’t all that uncommon, I suppose you expect me to first sign up of 10 emails and apply for 10 heroku accounts?

**Comment by fourjr** [comment]
50 servers? what kind of thing are you even running?

**Comment by kyb3r** [comment]
You can surely make changes [here](https://github.com/kyb3r/modmail/blob/master/core/thread.py#L297) to allow at least basic support for multiple guilds.

**Comment by WebKide** [comment]
Working on this, I'll keep you posted as new development comes available on the official repo

**Comment by WebKide** [comment]
**!-- FIRST CASE SCENARIO**
```
1) user messages modmail
2) modmail shares single server with user, then messages there to the mods
3) everything goes on as usual
```

**!-- SECOND CASE SCENARIO**
```
1) user messages modmail
2a) modmail shares 2 or more servers with user
2b) modmail acknowledges message received and asks which server user wants to message
2c) modmails gives options with names, maybe reacting or sending a number [or guild name]
3) after user answers, modmail starts the ticket where it belongs
4a) user's DMs are linked to that server until ticket is closed by Mods
4b) maybe user can also request to finish the ticket directly in DM to contact a different server's mods
```

**Comment by WebKide** [comment]
One more thought came to mind

If user shares a few servers with modmail and a few mods. How about instead of replying with Discord name, rather use user nick of the mod in the server they are replying?

Personally I have several nicks in various servers, and though people know it's me, the interactions tend to change based on the nick they see

**Comment by iamluca** [comment]
I think this feature should be supported not sure why you aren't supporting it - it will give you more users to use the bot and source code.

**Comment by maidos** [comment]
i like second case solution by webkide.
If bot detect that the recipient and bot share same server it will just give list which they are contacting from.

In order to relay your message to the right server please input the following option of which server you are contacting from:
Server A
Server B 
Server C

And then have the bot addreaction emojis A B C 
Once user react to either of it then the server thread creation message popup

**Comment by kyb3r** [comment]
Adding a ?newthread command or clickable emoji to start a thread (#224) would resolve which guild a thread is coming from.
